src/symmetric_heap_c.c

```diff
@@ -396,3 +399,27 @@ void SHMEM_FUNCTION_ATTRIBUTES * shmemalign(size_t alignment, size_t size)
 {
     return shmem_align(alignment, size);
 }
+
+
+void SHMEM_FUNCTION_ATTRIBUTES *
+shmemx_malloc_with_hints(size_t size, long hints)
+{
+    void *ret = NULL;
+
+    SHMEM_ERR_CHECK_INITIALIZED();
+
+    if (size == 0) return ret;
+
+    // Check for valid hints
+    if(hints >= SHMEMX_MALLOC_MAX_HINTS || hints < 0) {
+        RAISE_WARN_MSG("%ld is an invalid hint for shmem_malloc_with_hints. Valid hints can be 0, SHMEMX_MALLOC_ATOMICS_REMOTE, SHMEMX_MALLOC_SIGNAL_REMOTE or a combination of these three\n", hints);
```

Comment:
```suggestion
        RAISE_WARN_MSG("Ignoring invalid hint for shmem_malloc_with_hints (%ld)\n", hints);
```
If it was me, I would use the above. The existing message is too long to fit on a line (you can break it, there is a macro for the leading whitespace to align multi-line warnings) but also it will eventually go out of sync with what hints are allowed.

Reply:
aah got it! I did thought that this will seem like long for a warning.